### PR TITLE
fix hyperband, logger, search_method

### PR DESCRIPTION
**Pull request by loxacom123**

### CODE REVIEW

**Review by NoamRosenberg — COMMENTED** [review]
@loxacom123 could you elaborate? Under what conditions does value is None?

**Comment by loxacom123** [comment]
When search_method==random && (learning_rate * anchor_scales * anchor_ratios) > max_trials ,
populate_space return None ,that means all combinations have been tried.

File: hyperparameter_tuner/oracle.py [hunk 1]
```diff
@@ -28,9 +28,14 @@ def create_trial(self):
             values = None
         else:
             response = self._populate_space(trial_id)
-            status = response['status']
-            values = response['values'] if 'values' in response else None
-            self.trials[trial_id] = {'hp_values': values}
+            if response is None:
```
Comment (NoamRosenberg) [inline hyperparameter_tuner/oracle.py:31]: @loxacom123 can you elaborate on why this might happen?

File: hyperparameter_tuner/oracle.py [hunk 2]
```diff
@@ -28,9 +28,14 @@ def create_trial(self):
...
+                status = 'STOPPED'
+                values = None
+            else:
+                status = response['status']
+                values = response['values'] if 'values' in response else None
+                if values is not None:
```
Comment (NoamRosenberg) [inline hyperparameter_tuner/oracle.py:37]: @loxacom123 can you elaborate on why this might happen?

File: hyperparameter_tuner/oracle.py (hunk 1 above)
Comment (loxacom123) [reply hyperparameter_tuner/oracle.py:31]: When **search_method** set **random** and (**learning_rate** * **anchor_scales** * **anchor_ratios**) bigger than **max_trials** ,
**populate_space** return **None** ,that means all combinations have been tried.

Comment (loxacom123) [reply hyperparameter_tuner/oracle.py:31]: For instance,
``` json
"hp_search_space": [
      {
        "name": "learning_rate",
        "values": [
          5e-4,
          1e-5,
          5e-5
        ]
      },
      {
        "name": "anchor_scales",
        "values": [
          [1.0, 1.189207115002721, 1.4142135623730951],
          [1.0, 1.2599210498948732, 1.5874010519681994],
          [1.0, 1.5, 2.0]
        ]
      },
      {
        "name": "anchor_ratios",
        "values": [
          [0.306, 0.638, 1.0, 1.567, 3.266],
          [0.3, 0.5, 1.0, 2.0, 3.0],
          [0.6666666666666666, 0.8, 1.0, 1.25, 1.5]
        ]
      }
    ]
```
```learning_rate``` * ```anchor_scales``` * ```anchor_ratios```  = **27**
So when ```max_trials``` bigger than **27** , ```hyperparamter_tuner.oracle._populate_space()``` will return **None**.
So the ```status``` set as **STOPPED**

File: hyperparameter_tuner/oracle.py (hunk 2 above)
Comment (loxacom123) [reply hyperparameter_tuner/oracle.py:37]: When ```search_method``` set **hyperband** and ```_current_bracket_num``` = **0**,

the ```hyperparameter_tuner.hyperband._populate_space``` will return ``` {'status': 'STOPPED'}``` .

So the ```response['values']``` = **None** 

and the ```trials``` would be added a blank item.

Then ```hyperparameter_tuner.core.get_sorted_trial_ids()```  comes out error.

File: hyperparameter_tuner/oracle.py (hunk 1 above)
Comment (NoamRosenberg) [reply hyperparameter_tuner/oracle.py:31]: WELL PUT!

Comment (NoamRosenberg) [reply hyperparameter_tuner/oracle.py:31]: @loxacom123 thank you for this very elaborate explanation!

File: hyperparameter_tuner/oracle.py (hunk 2 above)
Comment (NoamRosenberg) [reply hyperparameter_tuner/oracle.py:37]: Well done @loxacom123!

**Pull request merged**